subprojects/jvm-services/src/test/groovy/org/gradle/jvm/toolchain/internal/MavenToolchainsInstallationSupplierTest.groovy

```diff
@@ -171,6 +171,86 @@ class MavenToolchainsInstallationSupplierTest extends Specification {
         useProperty << [true, false]
     }
 
+    def "supplies multiple paths for a single toolchain with resolvable environment variable"(boolean useProperty) {
+        given:
+        def toolchain = '''<toolchains>
+                                <toolchain>
+                                <type>jdk</type>
+                                <configuration>
+                                <jdkHome>${env.JDK16}</jdkHome>
+                                <jdkHome>${env.JDK17}</jdkHome>
```

Comment:
:x: I don't think having two `jdkHome` nodes inside a `configuration` is valid in Maven.
And while our parsing is basic and allows it, let's have tests with "valid" `toolchains.xml`